File: C#/0409. Longest Palindrome.cs
public class Solution {
    public int LongestPalindrome(string s) {
        if(s.Length==0){
            return 0;
        }
        Dictionary<char,int> sDict=new Dictionary<char,int>();
        foreach(char c in s){
            if(!sDict.ContainsKey(c)){
                sDict[c]=1;
            }
            else{
                sDict[c]+=1;
            }
        }
        int rep=0;
        int tmp=0;
        foreach(int sDictValue in sDict.Values){
            rep+=(sDictValue/2)*2;
            tmp+=sDictValue%2;
        }
        return tmp>0?rep+1:rep;
    }
}

File: 20190519python/0409. Longest Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        if not s:
            return 0
        
        s_dict={}
        for c in s:
            if c not in s_dict:
                s_dict[c]=1
            else:
                s_dict[c]+=1
        rep=0
        for k in s_dict:
            rep+=s_dict[k]//2
            
        if rep*2==len(s):
            return rep*2
        return rep*2+1
        
